src/lio/os/file.c

```diff
@@ -1128,7 +1128,7 @@ int va_timestamp_set_attr(lio_os_virtual_attr_t *va, lio_object_service_fn_t *os
     n = (int)(long)va->priv;  //** HACKERY ** to get the attribute prefix length
     key = &(fullkey[n+1]);
 
-    if (strlen(fullkey) < n) {  //** Nothing to do so return;
+    if ((int)strlen(fullkey) < n) {  //** Nothing to do so return;
```

Comment:
The C behavior around signed and unsigned comparisons and implicit conversions is damned bonkers.

```c
unsigned int x = 1;
int y = -1;
// Surprisingly false
return (x > y);
```

On the other hand, this gets the conversions right

```c
return (x - y > 0);
```

... go figure. In this case, though, it probably makes sense to have `n` be defined as `unsigned int`, right? I wasn't able to sort out what `n = (int)(long)va->priv;` actually resolved to, but if it's a length, changing the type is both more clear to the humans and sidesteps the compiler being truly surprising.

(I'm also curious about what the double conversion does at the end of the day. Does the compiler treat `(int)(long)` differently than `(int)`? Or does it optimize it all away?)